### PR TITLE
Improve RT alignment in feature finding by using only the middle 60% of each replicate…

### PR DESCRIPTION
**Pull request by bspratt**
…, avoiding the noise typically found at start and end of a mass spec run.

### CODE REVIEW

File: pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs [hunk 1]
```diff
@@ -235,77 +303,69 @@ private void TraceSouthWest(float[,] histogram, int bestXi, int bestYi, LinkedLi
...
-
         private void Stamp(float[,] histogram, float[,] stamp, int x, int y)
         {
-            var stampLength0 = stamp.GetLength(0); // GetLength is more expensive than you'd think
-            var stampLength1 = stamp.GetLength(1);
-            var histogramLength0 = histogram.GetLength(0);
-            var histogramLength1 = histogram.GetLength(1);
-
-            for (int i = x - stampLength0/2; i <= x + stampLength0 / 2; i++)
+            for (int i = x - stamp.GetLength(0) / 2; i <= x + stamp.GetLength(0) / 2; i++)
```
Comment (nickshulman) [inline pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs:332]: Does the "GetLength is more expensive than you'd think" comment still apply, or has more recent profiling disproven it?

File: pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs [hunk 2]
```diff
@@ -73,56 +76,95 @@ public KdeAligner(int origXFileIndex, int origYFileIndex, int resolution = 1000)
...
         }
 
-        public KdeAligner(int resolution = 1000) 
+        /// <summary>
+        /// Constructs a KdeAligner with a specific resolution and stretchFactor
+        /// </summary>
+        /// <param name="resolution">The number of points in the X and Y axes </param>
+        /// <param name="eccentricity">Experimental parameter controlling the amount that the CosineGaussian stamp is stretched.
+        /// This value is interpreted as the eccentricity of an ellipse whose major axis is along the Y=X axis.</param>
+        public KdeAligner(int resolution = 1000, double eccentricity = 0)
```
Comment (nickshulman) [inline pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs:85]: This "eccentricity" parameter should be removed. It was a misguided thing that I thought might help some scenario but never improved things.

File: pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs (hunk 1 above)
Comment (bspratt) [reply pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs:332]: I haven't reprofiled it. But I can remove the comment if it's distracting.

File: pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs (hunk 2 above)
Comment (bspratt) [reply pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs:85]: Will do, thanks

File: pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs (hunk 1 above)
Comment (bspratt) [reply pwiz_tools/Skyline/Model/RetentionTimes/KdeAligner.cs:332]: Oh right - uninitentional undoing of an optimization, thanks. Dang, I love your reviews.

**Pull request merged**